src/hooks/useEagerConnect.ts

```diff
@@ -0,0 +1,27 @@
+import { useWeb3React } from '@web3-react/core'
+import { useEffect, useState } from 'react'
+
+export default function useEagerConnect(
+  connect: (walletType: string) => Promise<void>
+) {
+  const { active } = useWeb3React()
+  const [tried, setTried] = useState(false)
+
+  // Connect to existing injected provider if app was opened from ONTO mobile wallet
+  useEffect(() => {
+    if (!active && !tried && (window as any).ethereum.isONTO) {
```

Comment:
`(window as any)` is obviously quite ugly. Any ideas on who I can avoid this ?

Reply:
Testing this locally it works, and removing the casting (like Uniswap has it) won't compile. I think we can go ahead with this but want to find a way to clean that bit up in the future